### PR TITLE
Fix video ui

### LINKED ISSUE #2427: [Regression] screenshot button pops out in the wrong place during video call
screenshot button

Reproducible: Always

Steps to reproduce:
1. Make video call
2. Move cursor over the file sending button

### PR DESCRIPTION
**Pull request by Diadlo**
Fix #2427 and #3396

### CODE REVIEW

File: src/widget/contentdialog.cpp [hunk 1]
```diff
@@ -360,6 +362,28 @@ void ContentDialog::cycleContacts(bool forward, bool loop)
...
+    videoCount++;
+    if (videoCount > 1)
+        return;
+
+    videoSurfaceSize = size;
+    QSize minSize = minimumSize();
+    setMinimumSize(minSize + videoSurfaceSize);
+}
+
+void ContentDialog::onVideoHiden()
```
Comment (antis81) [inline src/widget/contentdialog.cpp]: ~~Typo -> onVideoHidden~~ See above, please.

File: src/widget/contentdialog.cpp [hunk 2]
```diff
@@ -360,6 +362,28 @@ void ContentDialog::cycleContacts(bool forward, bool loop)
     }
 }
 
+void ContentDialog::onVideoShowed(QSize size)
```
Comment (antis81) [inline src/widget/contentdialog.cpp]: It's "onVideoShown". Why not simply "onVideoShow / onVideoHide" -> Less confusion.

File: src/widget/form/genericchatform.cpp [hunk 3]
```diff
@@ -25,11 +25,13 @@
 #include <QShortcut>
 #include <QKeyEvent>
 #include <QSplitter>
+#include <QPushButton>
```
Comment (antis81) [inline src/widget/form/genericchatform.cpp:26]: I tend to sort in alphabetical order and group by framework/namespace, as you did… makes things easier. So it's one line up in this case. Feel free to sort all includes in this file… :)

Comment (Diadlo) [inline src/widget/form/genericchatform.cpp:26]: > Feel free to sort all includes in this file

:smirk:
Sorted

File: src/widget/contentdialog.cpp (hunk 1 above)
Comment (Diadlo) [inline src/widget/contentdialog.cpp]: Fixed both

**Comment by anoadragon453** [comment]
For me this fixes the screenshot button as well as the chat area shrinking, but the chat log still goes behind the chat area.

**Comment by zetok** [comment]
```
$ echo -n 'fix(genericnetcamview, contentdialog, genericchatform): Changed minimum window size with video' | wc -c 
94
```

Um, no. Don't abuse `(scope)` not having a size limit. It's preferable to keep scope as short as possible – mentioning every file touched is not the purpose of scope. In this case, scope like `(video)` IMO would be perfectly fine.

**Comment by Diadlo** [comment]
@anoadragon453 Hmm... I can't reproduse. You do you resize? Window or video surface?

**Comment by anoadragon453** [comment]
Window:

screenshot_2016-07-03_02-14-19

Also Hide Messages button goes under/gets cut off by video view:

screenshot_2016-07-03_02-14-00

**Comment by Diadlo** [comment]
@anoadragon453 Fixed

**Comment by anoadragon453** [comment]
@Diadlo Works great, good job.

screenshot_2016-07-05_17-05-27

**Comment by anoadragon453** [comment]
This does seem to break the blue coloring in `/me` chat messages.

screenshot_2016-07-05_17-35-31

**Comment by Diadlo** [comment]
@anoadragon453 I can reproduce it on a01293f
Seems, it's regression

**Comment by zetok** [comment]
@Diadlo um, gh automatically turns commit hashes into links (if they're available in repo, that is), there's no need to do anything manually.

**Pull request merged**